README.md

```diff
@@ -37,6 +37,37 @@ within the `modules` directory, and run:
 pip install -e .
 ```
 
+### Noxfile
+
+[Nox](https://nox.thea.codes/en/stable/) is used to automate various development tasks running in isolated python environments.
+The following nox sessions are provided:
+
+- `test`: run tests for all supported python versions
+- `lint`: run `black` and `pylint` for all supported python versions
+- `type_check`: run `mypy` for all supported python versions
+
+To run a session, install the `nox` command (e.g., using pipx) and run:
+
+```shell
+nox -s <session_name>
+```
+
+To save time, reuse the session virtual environment using the `-r` option, i.e. `nox -rs <session_name>`.
+The checks are designed to conform to
+
+### Pre-commit
+
+[Pre-commit](https://pre-commit.com/) can be used to run a configured set of git pre-commit hooks before each commit. This is recommended.
+To set up the pre-commit hooks, install `pre-commit` (e.g., using pipx) and run:
```

Comment:
Do you mean pip?

Reply:
No, I mean [pipx](https://pypa.github.io/pipx/). It's a tool that works like pip but installs things in their own separate environment. It is good practice for installing python based command line tools (such as `nox` and `pre-commit`)